### ISSUE TITLE
[Fork-Join] Need improvement to the error messages

### ISSUE THREAD

**Issue by dilinisg** [ballerinalang/ballerina#2761]
1. In the below code segment, the if condition **(test["HotelWorker"] != null)** is incorrect where HotelWorker is not defined. But it throws the below error which is not meaningful to the user.

`ballerina.lang.errors:Error, message: unknown error occurred`

2. In the same code segment if the reply statement not available for a resource, the same error will be thrown, which is again not clear to the user

3. Same error when the join statement have invalid workers
eg: join (some 1 Tour1, XXX) (map test)

```
fork {
            worker Tour1 {
                system:println("Inside Tour1 Worker");
                system:println("constant value is in tour 1 is " + index);
                message tour1 = {};
                messages:setStringPayload(tour1, "This is tour 1");
                tour1 -> fork;
            }
        } join (some 1 Tour1) (map test) {

            system:println("Inside JOIN 2");
            if (test["HotelWorker"] != null) {
                any[] abc;
                abc, _ = (any[])test["HotelWorker"];
                results[0] = (message)abc[0];
                system:println(results[0]);
                //reply results;
                messages:setStringPayload(result, "ssssss");
                reply result;
            }
        }
```

**Comment by dilinisg** [comment]
This is happening as there were errors before executing the reply statement. So Ballerina sends this error message to the client to ack the client. This behaviour is wrong as it confuses the user

**Comment by lafernando** [comment]
This is fixed in v0.94, where meaningful errors are logged in invalid conditions.
